llarp/service/outbound_context.cpp

```diff
@@ -528,7 +528,90 @@ namespace llarp
     {
       m_LastInboundTraffic = m_Endpoint->Now();
       m_GotInboundTraffic = true;
-      return m_Endpoint->HandleHiddenServiceFrame(p, frame);
+      if (frame.R)
+      {
+        // handle discard
+        ServiceInfo si;
+        if (!m_Endpoint->GetSenderFor(frame.T, si))
+          return false;
+        // verify source
+        if (!frame.Verify(si))
+          return false;
+        // remove convotag it doesn't exist
+        LogWarn("remove convotag T=", frame.T);
+
+        AuthResult result{eAuthFailed, "unknown reason"};
+
+        SharedSecret sessionKey{};
+        if (m_DataHandler->GetCachedSessionKeyFor(frame.T, sessionKey))
+        {
+          ProtocolMessage msg{};
+          if (frame.DecryptPayloadInto(sessionKey, msg))
+          {
+            if (msg.proto == eProtocolAuth and not msg.payload.empty())
+            {
+              result.reason = std::string{reinterpret_cast<const char*>(msg.payload.data()),
+                                          msg.payload.size()};
+            }
+          }
+        }
+
+        m_Endpoint->RemoveConvoTag(frame.T);
+        if (authResultListener)
+        {
+          switch (frame.R)
+          {
+            case 1:
+              result.code = eAuthRejected;
+              break;
+            case 3:
```

Comment:
could use a comment noting why `case 2` is (apparently) omitted.

Reply:
Or better: change them to be constexpr values rather than magic numbers.

Reply:
now uses an enum class